workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/request/ExecuteRequestExecutor.java

```diff
@@ -45,6 +52,30 @@ public void execute(ActivityExecutorContext<ExecuteRequest> execution) throws IO
     execution.setOutputVariables(outputs);
   }
 
+  private String encodeQueryParameters(String fullUrl) {
+    // splits by the first occurrence of "?" to get the query params part
+    String[] splitUrl = fullUrl.split("\\?", 2);
+    String encodedUrl = splitUrl[0];
+
+    MultiValueMap<String, String> queryParamsMap =
+        UriComponentsBuilder.fromUriString(fullUrl).build().getQueryParams();
+
+    if (!queryParamsMap.isEmpty()) {
+      log.info("Encoding query parameters with Standard UTF-8 format");
+
+      String rawQuery = queryParamsMap
+          .keySet()
+          .stream()
+          .map(key ->
+              key + "=" + URLEncoder.encode(queryParamsMap.get(key).get(0), StandardCharsets.UTF_8))
```

Comment:
so if we have multiple query params in the map, what becomes of the other ones?

if the query param is already encoded, what will it do?